src/main/scala/org/constellation/primitives/CheckpointBlock.scala

```diff
@@ -532,14 +532,13 @@ sealed trait CheckpointBlockValidatorNel {
   }
 
   def containsAlreadyAcceptedTx(cb: CheckpointBlock)(implicit dao: DAO): IO[List[String]] = {
-    val containsAccepted = cb.transactions.toList
-      .map(
-        t =>
-          dao.transactionService
-            .isAccepted(t.hash)
-            .map(b => (t.hash, b))
-      )
-      .sequence[IO, (String, Boolean)]
+    val containsAccepted = cb.transactions.toList.map { t =>
+      dao.transactionService.lookup(t.hash).map {
+        case Some(tx) if tx.cbBaseHash != cb.baseHash.some => (t.hash, true)
```

Comment:
It looks like some operations happen in the middle of `cb` acceptance. So maybe migrating  `pendingAcceptance` to become `StorageService` or wrapping part of a code with semaphore would be an better option than here, what do you think about fixing it at `CheckpointService` level ?

Reply:
IMO we should firstly try to wrap acceptance into one semaphore and verify if it would fix the issue. If so, we can remove the semaphore and make the code properly concurrent